src/mine/stats.cpp

```diff
@@ -463,8 +475,24 @@ void Stats::updateProgramStats(UID id, const Program& program,
   if (with_loop) {
     has_loop.insert(id);
   }
+  bool with_formula = !formula_str.empty();
   if (with_formula) {
     has_formula.insert(id);
+    // Try to parse the formula and convert to PARI and LEAN
+    FormulaParser parser;
+    Formula formula;
+    if (parser.parse(formula_str, formula)) {
+      // Try PARI conversion (with offset 0 and as_vector false as defaults)
+      PariFormula pari_formula;
+      if (PariFormula::convert(formula, 0, false, pari_formula)) {
+        has_pari.insert(id);
+      }
+      // Try LEAN conversion (with offset 0 and as_vector false as defaults)
+      LeanFormula lean_formula;
+      if (LeanFormula::convert(formula, 0, false, lean_formula)) {
```

Comment:
The offset is important and can influence the decision if a formula can be converted or not. Therefore, pass in the offset as argument to the updateProgramStats function.